### ISSUE TITLE
Allow empty quadrants to be hidden (was: allow 3-pane layout)

### ISSUE THREAD

**Issue by ikashnitsky** [rstudio/rstudio#2854]
It would be extremely useful to have an option to choose 3 pane layout, something like this

[![fig1][f1]][f1]  

but without the collapsed History pane.

I wrote a couple of words on this issue in a [blog post][post]

[f1]: https://ikashnitsky.github.io/images/180522/layout-annotated.png
[post]: https://ikashnitsky.github.io/2018/perfect-rstudio-layout/

**Comment by ronblum** [comment]
@ikashnitsky Just to make sure I understand the request: what you've created is sort of faking a three-pane layout; using your example, you'd like the history moved to the lower right pane together with "Everything else" so that the Source pane is truly the only thing on the right?

**Comment by rmflight** [comment]
This would be great. Basically could be enabled by allowing one of the panes to have a "NULL" option, and not have anything displayed, and make **all** of the windows able to be shared among any of the tabs, i.e. would be great to have any combination the user wishes, including having **nothing**.

**Comment by ikashnitsky** [comment]
@ronblum exactly!
In this "fake three-pane layout" the bottom-left pane has an irritating habit of self unfolding, e.g. after you maximize (SHIFT+CTRL+#) any pane.

**Comment by rmflight** [comment]
@ikashnitaky that is exactly what I noticed when I tried this out, and why I proposed the *simple* (at least I think it should be simple) solution of allowing any combination of things, including NOTHING.

**Comment by hammer** [comment]
Just want to +1 this feature request; on my 21:9 (ultrawide) monitor, I really want a true three-column layout of [source, console, everything else]!

**Comment by hammer** [comment]
yeah that's a good start, thanks @jjallaire!

**Comment by hammer** [comment]
Also just noticed this feature request is a subset of https://github.com/rstudio/rstudio/issues/2879

**Comment by hammer** [comment]
@jjallaire any way to force files to open into the popped-out source pane? Right now when I open a new file in this setup it creates a source pane w/in the main RStudio window, which I then have to drag over to the popped out separate source pane...

**Comment by jmcphers** [comment]
@jjallaire No, there isn't; as you pointed out RStudio will remember which files were in separate windows, and will re-open those separate windows. However when opening a file it'll always appear in the main window.

@hammer What gesture are you using to open files? I wonder if some slightly smarter defaults could make this work more smoothly.

**Comment by hammer** [comment]
@jmcphers clicking on the file name in the Files pane, usually

**Comment by jmcphers** [comment]
@hammer Have you tried using <kbd>Ctrl</kbd> + <kbd>.</kbd> while you have the popped-out source pane focused? Source windows have their own fuzzy finders, and if you use that gesture to open a file, the newly opened file will go into the popped-out window.

**Comment by hammer** [comment]
@jmcphers yes, that works! I may have to retrain my fingers but I can try to stick w/ that for now.

**Comment by hammer** [comment]
@jjallaire @jmcphers another issue I've encountered when using the popped-out source pane: the `citr` add-in seems to generate the citation in the terminal window even when I invoke it from the popped-out source pane.

**Comment by MilesMcBain** [comment]
I used to lay out my RStudio exactly as per the image. But as others have mentioned the panes have an annoying habbit of popping open. Pretty sure we had this raised as a bug at some point but it still seems to be happening.

I now use ESS where I have unlimted control of windows. I would say my window usage breakdown would be something like 80% 3 panes, 19% 2 panes, 1%  3+ panes.

Plots I always open in full screen on an alternative monitor or desktop. Rmd renders likewise.

**Comment by kiernann** [comment]
Similarly, I'd love a feature like this that might allow for a (1) full-width console on bottom, (2) half-width environment and git panes above it, (3) and a separate source editor window.

A full-width console is particularly important when using tibbles, which smartly limit row printing but benefit from more space for columns.

**Comment by cppcooper** [comment]
Yes to more pane options. I would like to have a :point_up: full width pane across the top and my env and files on the bottom (side by side) because I want to keep RStudio on my portrait monitor. Would be ideal to also have the console and source in the same pane (for me), but that is kind of off topic.

**Comment by salim-b** [comment]
I think the requested feature is provided by RStudio 1.4's [multiple source pane feature](https://community.rstudio.com/t/new-feature-editing-with-multiple-source-columns/80589). Downloads of RStudio 1.4 daily builds (considered unstable) are available [here](https://dailies.rstudio.com/).

So I guess this issue can be closed.

**Comment by francisbarton** [comment]
> 
> 
> I think the requested feature is provided by RStudio 1.4's [multiple source pane feature](https://community.rstudio.com/t/new-feature-editing-with-multiple-source-columns/80589). Downloads of RStudio 1.4 daily builds (considered unstable) are available [here](https://dailies.rstudio.com/).
> 
> So I guess this issue can be closed.

It's close, but I don't agree that the new source pane feature meets exactly what the OP here was asking for.

**Comment by cppcooper** [comment]
Yup I agree, to me this feature request represents the ability to remove or disable existing panes and perhaps even choose two or more panes to merge.

**Comment by jucor** [comment]
Loving the new extra source pane in 1.4. One feature would make it much more useful: the ability to open a same file in both columns, please :) This would essentially allow what vim's split-panes does: seeing multiple parts of the same file at once.
Thanks for considering!

**Comment by verajosemanuel** [comment]
Really annoying that the bottom left pane keeps unfolding. I only need two panes: source and viewer, but I am unable to set a stable layout with this requisites.
